spec/models/service_spec.rb

```diff
@@ -732,6 +732,42 @@
     end
   end
 
+  describe '#add_to_service' do
+    let(:service) { FactoryGirl.create(:service) }
+    let(:child_service) { FactoryGirl.create(:service) }
+
+    it 'associates a child_service to the service' do
+      child_service.add_to_service(service)
+
+      expect(service.reload.services).to include(child_service)
+    end
+
+    it 'raise an error if the child_service is already part of a service' do
+      child_service.add_to_service(service)
+
+      expect { child_service.add_to_service(service) }.to raise_error MiqException::Error
+    end
+  end
+
+  describe '#remove_from_service' do
+    let(:service) { FactoryGirl.create(:service) }
+    let(:child_service) { FactoryGirl.create(:service) }
+
+    it 'associates a child_service to the service' do
+      child_service.add_to_service(service)
+      expect(child_service.service).to eq(service)
+
+      child_service.remove_from_service(service)
+      expect(child_service.service).to be_nil
```

Comment:
Should it be ```expect(service.services).to be_nil```?

Reply:
Same as above, I think both are valid.  But here since we are checking that the removal logic worked it might be worth it to check both conditions.  I'll push an update and you can let me know what you think.